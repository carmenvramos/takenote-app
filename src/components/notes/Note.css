.Note {
  display: block;
  margin: auto;

  h2 {
    padding: 0.4em 0;
    color: #B16FB1;
  }

  textarea {
    line-height: 1.5em;
  }
}  