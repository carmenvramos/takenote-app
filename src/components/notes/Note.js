import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { FaPen } from 'react-icons/fa';
import styles from './Note.css';

function Note({ note }) {
  const { _id, title, body, author } = note;
  return (
    <article className={styles.Note}>
      <h2>{title}</h2>
      <h5>by {author}</h5>
      <p>{body}</p>
      <Link to={`/notes/${_id}/edit`}>
        <FaPen />
      </Link>
    </article>
  );
}

Note.propTypes = {
  note: PropTypes.object.isRequired,
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired
};

export default Note;
