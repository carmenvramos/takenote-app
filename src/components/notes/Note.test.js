import React from 'react';
import { shallow } from 'enzyme';
import Note from './Note';

describe('Note tests', () => {
  it('matches a snapshot', () => {
    const note = {
      _id: '11',
      author: 'test.user',
      title: 'Note Title',
      body: 'Note Body'
    };
    expect(shallow(
      <Note note={note} />
    )).toMatchSnapshot();
  });
});
