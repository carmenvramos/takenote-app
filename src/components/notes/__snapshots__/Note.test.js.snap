// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note tests matches a snapshot 1`] = `
<article
  className="Note"
>
  <h2>
    Note Title
  </h2>
  <h5>
    by 
    test.user
  </h5>
  <p>
    Note Body
  </p>
  <Link
    replace={false}
    to="/notes/11/edit"
  >
    <FaPen />
  </Link>
</article>
`;
